### PR TITLE
Restore tagline to MainResponse in server and RHLC

### LINKED ISSUE #901: [BUG] Elasticsearch RestHighLevelClient can not get response of Info API from OpenSearch server due to missing "tagline" field in MainResponse
**Describe the bug**

Source: https://discuss.opendistrocommunity.dev/t/spring-data-integration/6239/4

Because the "tagline" field is removed from "MainResponse" in both sever and RHLC side of OpenSearch in PR https://github.com/opensearch-project/OpenSearch/pull/427, [Info API](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/7.10/java-rest-high-main.html) calls from the existing Elasticsearch Rest-High-Level-Client (such as version 7.10.2) can not get the required "tagline" field from OpenSearch server (version 1.0.0). It causes compatibility issues with some libraries that support Elasticsearch.

For example, [SpringData for Elasticsearch](https://github.com/spring-projects/spring-data-elasticsearch) uses [elasticsearch-rest-high-level-client](https://github.com/spring-projects/spring-data-elasticsearch/blob/4.1.x/pom.xml#L166) library to call `org.elasticsearch.client.RestHighLevelClient.info()` to retrieve the version of the ES cluster in this [line](https://github.com/spring-projects/spring-data-elasticsearch/blob/84a74f5921c7f5c0939d6f0fc938bba1bef4e5d3/src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java#L373).
"tagline" is a required field in the response of that call, without the field, there will be an error.

**To Reproduce**
No detailed steps, but use "SpringData for Elasticsearch" with an OpenSearch 1.0.0 cluster.

**Expected behavior**
- "SpringData for Elasticsearch" version 4.x can be compatible with OpenSearch 1.0.0 with the functionality made for Elasticsearch 7.10.2 .
- Anything that uses the [Elasticsearch RHLC library](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/7.10/java-rest-high-getting-started-maven.html#java-rest-high-getting-started-maven-maven) can get response of "Info API" from OpenSearch 1.0.0 clusters.

**Plugins**
No plugins enabled.

**Screenshots**
(none)

**Host/Environment (please complete the following information):**
Platform-independent 

**Additional context**
The issue of removing the tagline: https://github.com/opensearch-project/OpenSearch/issues/362
The issue of requesting support for "Spring Data for Elasticsearch": https://github.com/opensearch-project/OpenSearch/issues/542

### PR DESCRIPTION
**Pull request by tlfeng**
### Description
- Revert PR #427 to add "tagline" field back to "MainResponse" in both sever and rest-high-level-client sides.
- Replace with a new tagline "The OpenSearch Project: https://opensearch.org/".
- Turn the tagline into a constant in `server/src/main/java/org/opensearch/action/main/MainResponse.java`.
 
**Note**: Adding the tagline back to RestHighLevelClient side will be a breaking change to prevent OpenSearch RHLC "post RC-1" calling Info API from OpenSearch sever before 1.0.0-rc1.

Current output of the MainResponse:
```
{
  "name" : "node-9200",
  "cluster_name" : "opensearch",
  "cluster_uuid" : "em8tu3VZQCWvzGnouyZVpg",
  "version" : {
    "distribution" : "opensearch",
    "number" : "1.1.0-SNAPSHOT",
    "build_type" : "deb",
    "build_hash" : "1896ab72680760923a85db99a20c42b75408e8f6",
    "build_date" : "2021-06-30T22:07:19.423955Z",
    "build_snapshot" : true,
    "lucene_version" : "8.8.2",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  }
}
```

After the change in the PR:
```
{
  "name" : "node-9200",
  "cluster_name" : "opensearch",
  "cluster_uuid" : "IoY5158sQp6oz5lkCPCVUg",
  "version" : {
    "distribution" : "opensearch",
    "number" : "1.1.0-SNAPSHOT",
    "build_type" : "tar",
    "build_hash" : "79e1180251d26fd98fa88e42d4c4eca1a6768b09",
    "build_date" : "2021-06-30T23:15:13.489461Z",
    "build_snapshot" : true,
    "lucene_version" : "8.8.2",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
```

### Issues Resolved
Resolve #901 
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
start gradle check

File: server/src/main/java/org/opensearch/action/main/MainResponse.java [hunk 1]
```diff
@@ -149,6 +149,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             .field("minimum_wire_compatibility_version", version.minimumCompatibilityVersion().toString())
             .field("minimum_index_compatibility_version", version.minimumIndexCompatibilityVersion().toString())
             .endObject();
+        builder.field("tagline", "The OpenSearch Project: https://opensearch.org/");
```
Comment (abbashus) [inline server/src/main/java/org/opensearch/action/main/MainResponse.java:152]: I feel the value should be turned into constant.

**Review by abbashus — APPROVED** [review]
One comment. Else looks good to me.

File: server/src/main/java/org/opensearch/action/main/MainResponse.java (hunk 1 above)
Comment (tlfeng) [reply server/src/main/java/org/opensearch/action/main/MainResponse.java:152]: I agree with it... I can take some time in finding a good place for the constant.

Comment (tlfeng) [reply server/src/main/java/org/opensearch/action/main/MainResponse.java]: Made the tagline become a constant in this class. Thank you!

**Comment by abbashus** [comment]
start gradle check

**Comment by tlfeng** [comment]
The above test failure is:
```
  2> REPRODUCE WITH: ./gradlew ':modules:transport-netty4:test' --tests "org.opensearch.transport.netty4.SimpleNetty4TransportTests.testHandshakeUpdatesVersion" -Dtests.seed=59DADD0DFC828D39 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=en-AU -Dtests.timezone=Etc/GMT-14 -Druntime.java=15
  2> java.lang.AssertionError: expected:<7.10.2> but was:<1.1.0>
        at __randomizedtesting.SeedInfo.seed([59DADD0DFC828D39:E8996B23DD77BEC4]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.opensearch.transport.AbstractSimpleTransportTestCase.testHandshakeUpdatesVersion(AbstractSimpleTransportTestCase.java:2016)
...
```

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
Test failure in Log 276:
```
> Task :test:fixtures:minio-fixture:composeUp

ERROR: for 81df22525be5077429a7afdcff93efee_minio-fixture__minio-fixture-for-snapshot-tool_1  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
Creating 81df22525be5077429a7afdcff93efee_minio-fixture__minio-fixture-other_1             ... done
Creating 81df22525be5077429a7afdcff93efee_minio-fixture__minio-fixture_1                   ... done

ERROR: for minio-fixture-for-snapshot-tool  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
An HTTP request took too long to complete. Retry with --verbose to obtain debug information.
If you encounter this issue regularly because of slow network conditions, consider setting COMPOSE_HTTP_TIMEOUT to a higher value (current value: 60).
```
```
* What went wrong:
Execution failed for task ':test:fixtures:minio-fixture:composeUp'.
> Exit-code 1 when calling /usr/bin/docker-compose, stdout: N/A
```

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
Test failure in Log 277:
```
> Task :qa:mixed-cluster:v1.0.0#mixedClusterTest

REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v1.0.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=info/10_info/Info}" -Dtests.seed=C5A52FCEF7CBB11F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=is-IS -Dtests.timezone=Africa/Monrovia -Druntime.java=15

org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=info/10_info/Info} FAILED
    java.lang.AssertionError: Failure at [info/10_info:7]: field [tagline] doesn't have a true value
    Expected: not null
         but: was null
```

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
Test failure in Log 280:
```
> Task :server:internalClusterTest

REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.action.support.master.IndexingMasterFailoverIT.testMasterFailoverDuringIndexingWithMappingChanges" -Dtests.seed=B6525E9D42487AD6 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=uk-UA -Dtests.timezone=Etc/GMT+8 -Druntime.java=15

org.opensearch.action.support.master.IndexingMasterFailoverIT > testMasterFailoverDuringIndexingWithMappingChanges FAILED
    java.lang.AssertionError: 
    Expected: <10L>
         but: was <0L>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.opensearch.action.support.master.IndexingMasterFailoverIT.testMasterFailoverDuringIndexingWithMappingChanges(IndexingMasterFailoverIT.java:127)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=835, name=indexingThread, state=RUNNABLE, group=TGRP-IndexingMasterFailoverIT]

        Caused by:
        MasterNotDiscoveredException[null]
```

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
Test failure in Log 281:
```
> Task :qa:rolling-upgrade:v1.0.0#twoThirdsUpgradedTest

REPRODUCE WITH: ./gradlew ':qa:rolling-upgrade:v1.0.0#twoThirdsUpgradedTest' --tests "org.opensearch.upgrades.IndexingIT.testAutoIdWithOpTypeCreate" -Dtests.seed=A1D319810F427F3F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1" -Dtests.locale=es-MX -Dtests.timezone=Pacific/Niue -Druntime.java=15

org.opensearch.upgrades.IndexingIT > testAutoIdWithOpTypeCreate FAILED
    java.lang.AssertionError: there are still running tasks:
    {time_in_queue=92ms, time_in_queue_millis=92, source=shard-started StartedShardEntry{shardId [[test-auto-expand-filtering][0]], allocationId [RppMT-9tQ3i2ASu6VIDCWw], primary term [2], message [after peer recovery]}, executing=true, priority=URGENT, insert_order=77}
```

**Comment by tlfeng** [comment]
start gradle check

**Comment by tlfeng** [comment]
The test failure in Gradle check log 277 is reasonable. BWC test using the latest snapshot of OpenSearch 1.0.0, where "tagline" is neither in MainResponse of sever side nor of rest-client side, so the test may fail when calling "Info API". So I created PR #913 to only restore "tagline" in sever side, but not in rest-client side, aim to pass the BWC test.

**Comment by nknize** [comment]
can you open the revert portion of the PR separate from the changes being made?

**Review by nknize — CHANGES_REQUESTED** [review]
can you open the revert portion of the PR separate from the changes being made so we can cleanly track history?

**Comment by tlfeng** [comment]
> can you open the revert portion of the PR separate from the changes being made so we can cleanly track history?

Hi Nick, sure, I totally agree with separating the revert portion with the new changes. 😄
But the problem is that reverting the whole PR # 427 will often fails our current BWC test, which prevent the merging.

**Comment by nknize** [comment]
> But the problem is that reverting the whole PR # 427 will often fails our current BWC test, which prevent the merging.

Ah, so you're saying this doesn't revert the whole PR #427?  If that's the case then just change the explanation here to something like `add "tagline" field back to "MainResponse" that was removed in #427`

**Comment by tlfeng** [comment]
> Ah, so you're saying this doesn't revert the whole PR #427? If that's the case then just change the explanation here to something like `add "tagline" field back to "MainResponse" that was removed in #427`

I see. Thank you for your suggestion! I wish I can revert whole first and then make new changes to make a clear history, but due to our BWC tests enabled, I think I can only add "tagline" back partially, which is not reverting the whole #427 anymore...

**Comment by tlfeng** [comment]
I will close the PR if adding "tagline" back to MainResponse in both sever and client side doesn't pass the test, and instead I will rename my new PR #913 .

**Comment by nknize** [comment]
> I wish I can revert whole first and then make new changes to make a clear history,

If you have an open PR to fix the BWC failures then if we can merge it right after it shouldn't cause any issue.

**Comment by tlfeng** [comment]
> If you have an open PR to fix the BWC failures then if we can merge it right after it shouldn't cause any issue.

Hi Nick, appreciate your help and your advice!
If the maintainers agree, I plan to go on with PR 913 and close this # 902. After the change in 913 get into 1.0 branch, I think I can add the tagline back to MainResponse in RHLC side without BWC test failure.

**Comment by nknize** [comment]
> I plan to go on with PR 913 and close this # 902

agree... closing this in favor of #913

**Pull request closed without merging**